library/ssl_srv.c

```diff
@@ -2588,36 +2588,45 @@ static int ssl_write_certificate_request( mbedtls_ssl_context *ssl )
      * opaque DistinguishedName<1..2^16-1>;
      */
     p += 2;
-#if defined(MBEDTLS_SSL_SERVER_NAME_INDICATION)
-    if( ssl->handshake->sni_ca_chain != NULL )
-        crt = ssl->handshake->sni_ca_chain;
-    else
-#endif
-        crt = ssl->conf->ca_chain;
 
     total_dn_size = 0;
-    while( crt != NULL && crt->version != 0 )
+
+#if defined(MBEDTLS_SSL_SRV_C)
```

Comment:
Since we're in `ssl_srv.c` I think this is redundant (same for the one on the closing brace obviously).